pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -360,6 +376,9 @@ func (r *ReconcileConstraintTemplate) handleUpdate(
 			log.Error(err, "conversion error")
 			return reconcile.Result{}, err
 		}
+		if err := controllerutil.SetControllerReference(instance, crdv1beta1, r.scheme); err != nil {
```

Comment:
This one is not quite right - it will only get set if the spec has changed, but `ownerReferences` are part of `objectMeta`. The way I generally approach this is by writing a function to generate the full desired state (in our case including the ownerReferences) - and then separating the logic of patching it into the cluster.

Reply:
I think I can get rid of the deepequal altogether here. IIUC changes to status will not be persisted on update for the CRD resource, correct?

Reply:
Yes I believe that is correct: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#status-subresource